### PR TITLE
[Group 1] Add Elevator Subsystem

### PR DESCRIPTION
**Pull request by vincentw921**
Co-authored-by: Kelvin Zhao <t1rbr@users.noreply.github.com>
Co-authored-by: Samuel Chen <SamChen1738@users.noreply.github.com>

### CODE REVIEW

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    private final Encoder grayhill;
+
+    private SmartNumber targetDistance;
+
+    private Mechanism2d elevator;
+
+    public Elevator() {
+        setSubsystem("Elevator");
+        WPI_TalonSRX firstMotor = new WPI_TalonSRX(Ports.Elevator.FIRST);
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:59]: Talons or CANSparkMax? Also configure motors

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    private final Controller controller;
+    private final DigitalInput lowerLimitSwitch;
+    private final DigitalInput upperLimitSwitch;
+
+    private final Encoder grayhill;
+
+    private SmartNumber targetDistance;
+
+    private Mechanism2d elevator;
+
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:56]: Author tags

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ */
+
+public class Elevator extends SubsystemBase {
+    
+    private final MotorControllerGroup motors;
+    private final Controller controller;
+    private final DigitalInput lowerLimitSwitch;
+    private final DigitalInput upperLimitSwitch;
+
+    private final Encoder grayhill;
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:51]: Ask if using grayhills

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        WPI_TalonSRX thirdMotor = new WPI_TalonSRX(Ports.Elevator.THIRD);
+        motors = new MotorControllerGroup(firstMotor, secondMotor, thirdMotor);
+        addChild("Motor Controller", motors);
+
+        controller = new PIDController(kP, kI, kD)
+                            .add(new Feedforward.Elevator(kG, kS, kV, kA).position());
+
+        lowerLimitSwitch = new DigitalInput(Ports.Elevator.LOWER);
+        addChild("Lower Limit Switch", lowerLimitSwitch);
+        upperLimitSwitch = new DigitalInput(Ports.Elevator.UPPER);
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:70]: Limit switches never implemented

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        addChild("Upper Limit Switch", upperLimitSwitch);
+        grayhill = new Encoder(Ports.Elevator.A, Ports.Elevator.B);
+        addChild("Encoder", grayhill);
+
+        targetDistance = new SmartNumber("Elevator/Target Distance", 0.0);
+        
+        elevator = new Mechanism2d(WIDTH, HEIGHT);
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:77]: Implement simulations for elevator

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        targetDistance.set(Settings.Elevator.SWITCH);
+    }
+
+    public void setClimb() {
+        targetDistance.set(Settings.Elevator.RUNG);
+    }
+
+    @Override
+    public void periodic() {
+        motors.set(controller.update(targetDistance.get(),
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:106]: change set to setVoltage

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java (hunk 1 above)
Comment (selym3) [reply Group1/src/main/java/com/stuypulse/robot/subsystems/Elevator.java:59]: yea

**Pull request merged**